django_lti_tool_provider/views.py

```diff
@@ -45,10 +46,19 @@ def post(self, request, *args, **kwargs):
         return self.process_request(request)
 
     def process_request(self, request):
+        if request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                if not self._right_user(request.user, lti_parameters):
+                    _logger.debug(u"Logging out user %s in favor of new LTI session.", request.user.username)
+                    logout(request)
+            except (oauth2.Error, AttributeError):
+                # Not a new visit, or better to keep existing auth.
+                pass
```

Comment:
@Kelketek I'm not sure, but it doesn't look like it would.

* oauth error signifies there's a malformed oauth request - it is totally safe to throw 400 right away, and it needs to be logged, I believe. Actually, since process request is a single entry point to both anonymous and authenticated paths, it might make sense to do the parsing once, send bad request if oauth is failed, than just pass parsed values along; kind of "Fence"-pattern.
* Can you please explain why AttributeError is cathced here, but not WrongUserError?